Error in test cases (code coverage): Date 03-10-2025 is not in any active Fiscal Year

### Information about bug

Common error in multiple test cases from code coverage i.e. erpnext.accounts.utils.FiscalYearError: Date 03-10-2025 is not in any active Fiscal Year

test cases name:-
test_multi_payment_request_for_purchase_invoice_TC_ACC_036
test_over_billing_allowance_TC_ACC_119
test_partly_paid_of_pi_to_pr_to_pe_TC_B_081
test_partly_paid_of_pi_to_pr_to_pe_with_gst_TC_B_083
test_payment_term_discount_for_pi_at_fully_paid_TC_ACC_098
test_payment_term_discount_for_pi_at_partially_paid_TC_ACC_100
test_pi_standalone_pi_with_deferred_expense_TC_B_095
test_pi_with_additional_discount_TC_B_054
test_pi_with_additional_discount_TC_B_060
test_purchase_invoice_payment_and_cancel_invoice_TC_ACC_019
test_repost_account_ledger_for_pi_TC_ACC_117
test_single_payment_request_for_purchase_invoice_TC_ACC_035
test_standalone_pi_is_fully_paid_TC_B_088
test_standalone_pi_is_partly_paid_TC_B_090
test_stop_pi_creation_when_value_exceeds_budget_TC_ACC_133
test_supplier_invoice_number_uniqueness_validation_TC_ACC_136
test_tax_withholding_with_supplier_TC_ACC_023
test_warn_pi_creation_when_value_exceeds_budget_TC_ACC_145
test_calculate_commission_for_sales_partner_TC_ACC_143
test_create_sales_invoice_for_common_party_TC_ACC_124
test_create_sales_invoice_for_interstate_branch_transfer_TC_ACC_123
test_deferred_revenue_invoice_line_item_TC_ACC_039
test_deferred_revenue_invoice_multiple_item_TC_ACC_040
test_direct_sales_invoice_via_update_stock_TC_SCK_132
test_jv_records_creation_diff_ex_rate_TC_ACC_029
test_jv_records_creation_diff_ex_rate_TC_ACC_030
test_multiple_payment_request_for_purchase_invoice_TC_ACC_038
test_payment_term_discount_for_si_at_fully_paid_TC_ACC_097
test_payment_term_discount_for_si_at_partially_paid_TC_ACC_099
test_prevent_sale_below_purchase_rate_TC_ACC_125
test_repost_account_ledger_for_si_TC_ACC_118
test_sales_invoice_and_delivery_note_with_shipping_rule_TC_S_026
test_sales_invoice_creating_dunning_from_si_TC_S_154
test_sales_invoice_dont_reserve_sales_order_qty_on_sales_return_TC_S_158
test_sales_invoice_ignoring_pricing_rule_TC_S_156
test_sales_invoice_to_allow_item_multiple_times_TC_S_159
test_sales_invoice_with_SR_and_CRN_TC_S_038
test_sales_invoice_with_child_item_rates_of_product_bundle_TC_S_152
test_sales_invoice_with_sr_crn_and_payment_TC_S_039
test_sales_invoice_with_update_stock_and_SR_TC_S_027
test_sales_invoice_with_update_stock_checked_TC_S_007
test_sales_invoice_without_sales_order_TC_S_006
test_si_cancel_amend_with_customer_change_TC_S_129
test_si_cancel_amend_with_item_details_change_TC_S_128
test_si_cancel_amend_with_payment_terms_change_TC_S_130
test_si_credit_note_cancel_amend_with_payment_terms_change_TC_S_131
test_si_with_deferred_revenue_item_TC_S_135
test_si_with_sr_calculate_with_fixed_TC_S_139
test_si_with_sr_calculate_with_net_total_TC_S_140
test_si_with_sr_calculate_with_net_weight_TC_S_141
test_single_payment_request_for_purchase_invoice_TC_ACC_037
test_tax_with_holding_with_si_TC_ACC_109
test_test_unlink_payment_on_invoice_cancellation_TC_ACC_126
test_subscription_plan_with_template_for_pi_TC_ACC_086
test_subscription_plan_with_template_for_si_TC_ACC_087
test_create_tax_rule_and_apply_to_purchase_invoice_TC_ACC_102
test_create_tax_rule_and_apply_to_sales_invoice_TC_ACC_101
test_create_po_pr_return_pr_TC_SCK_178
test_create_material_req_to_2po_to_pi_serial_TC_SCK_096
test_create_mr_to_2po_to_1pi_serial_cancel_TC_SCK_099
test_create_mr_to_2po_to_1pi_serial_return_TC_SCK_111
test_create_mr_to_2po_to_1pi_sr_prtl_ret_TC_SCK_115


### Module

stock

### Version

Frappe version : pre-prod
ERPNext version: Pre-prod

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell

```

@ksagar8848 
test_standalone_pi_is_fully_paid_TC_B_088 ,test_standalone_pi_is_partly_paid_TC_B_090  -  ValidationError: A template with tax category _Test Tax Category 1 already exists. Only one template is allowed with each tax category

test_deferred_revenue_invoice_line_item_TC_ACC_039 ,test_direct_sales_invoice_via_update_stock_TC_SCK_132 ,test_sales_invoice_dont_reserve_sales_order_qty_on_sales_return_TC_S_158 , test_sales_invoice_with_SR_and_CRN_TC_S_038, test_sales_invoice_with_child_item_rates_of_product_bundle_TC_S_152 ,test_sales_invoice_with_sr_crn_and_payment_TC_S_039, test_sales_invoice_with_update_stock_and_SR_TC_S_027,test_sales_invoice_with_update_stock_checked_TC_S_007 - AttributeError: 'AssetsSalesInvoice' object has no attribute 'get_finished_item_row'

test_sales_invoice_without_sales_order_TC_S_006 - AttributeError: 'DeliveryNote' object has no attribute 'get_finished_item_row'

test_si_with_sr_calculate_with_fixed_TC_S_139 - AssertionError: 600.0 != 1050.0

test_tax_with_holding_with_si_TC_ACC_109 - AssertionError: 152772.0 != 151000.0

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -5302,6 +5302,12 @@ class TestSalesInvoice(FrappeTestCase):
 			account_currency="INR",
 			account_type="Cash",
 		)
+		create_account(
+			account_name="Deferred Revenue",
+			parent_account="Current Liabilities - _TC",
+			company="_Test Company",
+			account_currency="INR",
+		)
 
 		create_customer(
 			customer_name="_Test Customer",
@@ -5310,6 +5316,12 @@ class TestSalesInvoice(FrappeTestCase):
 			account="_Test Receivable - _TC"
 		)
 
+		frappe.set_value("Company", "_Test Company", {
+			"default_receivable_account": "Debtors - _TC",
+			"default_expense_account": "Cost of Goods Sold - _TC",
+			"default_inventory_account": "Stock In Hand - _TC",
+			"enable_perpetual_inventory":1
+		})
 		item = make_test_item(item_name="_Test Item")
 		item.enable_deferred_revenue=1
 		item.no_of_months=12
@@ -5353,6 +5365,7 @@ class TestSalesInvoice(FrappeTestCase):
                 ['Deferred Revenue - _TC', 0.0, sales_invoice.grand_total, sales_invoice.posting_date]
         ]
 		check_gl_entries(self, sales_invoice.name, expected_gl_entries, sales_invoice.posting_date)
+		frappe.db.rollback()
     
 	def test_deferred_revenue_invoice_multiple_item_TC_ACC_040(self):
 		from erpnext.accounts.doctype.cost_center.test_cost_center import create_cost_center
@@ -5957,40 +5970,27 @@ class TestSalesInvoice(FrappeTestCase):
 		self.assertEqual(si.status, "Unpaid")	
 
 	def test_si_with_sr_calculate_with_fixed_TC_S_139(self):
-		from erpnext.stock.doctype.stock_entry.test_stock_entry import make_stock_entry
-		from erpnext.buying.doctype.purchase_order.test_purchase_order import get_or_create_fiscal_year
 		from erpnext.accounts.doctype.shipping_rule.test_shipping_rule import create_shipping_rule
-		from erpnext.accounts.doctype.cost_center.test_cost_center import create_cost_center
-  
-		create_item(item_code = "_Test Item")
+		from erpnext.stock.doctype.stock_entry.test_stock_entry import make_stock_entry
+		from erpnext.stock.doctype.item.test_item import create_item
+		from erpnext.buying.doctype.purchase_order.test_purchase_order import get_or_create_fiscal_year
 		get_or_create_fiscal_year("_Test Company")
-		frappe.db.set_single_value("Selling Settings", "so_required", "No")
-		create_account(
-			account_name="_Test Account Shipping Charges",
-			parent_account="Cash In Hand - _TC",
-			company="_Test Company",
-			account_currency="INR",
-			account_type="Chargeable",
-		)
-		create_cost_center(cost_center_name="_Test Cost Center", company="_Test Company")
-		create_customer(customer_name="_Test Customer",company="_Test Company")
-
-		shipping_charge = 100
-		shipping_rule = create_shipping_rule(shipping_rule_type="Selling", shipping_rule_name="_Test Shipping Rule")
-  
-		make_stock_entry(item_code="_Test Item", qty=10, rate=500, target="_Test Warehouse - _TC")
-  
-		si = create_sales_invoice(qty=5,rate=200, shipping_rule="_Test Shipping Rule", do_not_submit=True)
+		item = create_item("_Test Item 1")
+		shipping_rule = create_shipping_rule(
+			shipping_rule_type="Selling", 
+			shipping_rule_name="Shipping Rule - Test Fixed",
+			args={"calculate_based_on": "Fixed", "shipping_amount": 100}
+    	)
+		self.assertEqual(shipping_rule.docstatus, 1)
+		make_stock_entry(item_code="_Test Item 1", qty=10, rate=500, target="_Test Warehouse - _TC")
+		si = create_sales_invoice(qty=5,rate=200, do_not_submit=True)
+		si.shipping_rule = shipping_rule.name
 		si.save()
 		si.submit()
-  
-		for ship in shipping_rule.conditions:
-			if ship.from_value <= si.items[0].qty <= ship.to_value:
-				shipping_charge = ship.shipping_amount
-
-		expected_grand_total = 1000 + shipping_charge
-
-		self.assertEqual(si.grand_total, expected_grand_total)
+		self.assertEqual(si.net_total, 1000)
+		self.assertEqual(si.total_taxes_and_charges, 100)
+		self.assertEqual(si.grand_total, 1100)
+		frappe.db.rollback()
 
 	def test_si_with_sr_calculate_with_net_total_TC_S_140(self):
 		from erpnext.accounts.doctype.shipping_rule.test_shipping_rule import create_shipping_rule

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -5976,6 +5976,7 @@ class TestSalesInvoice(FrappeTestCase):
 		from erpnext.buying.doctype.purchase_order.test_purchase_order import get_or_create_fiscal_year
 		get_or_create_fiscal_year("_Test Company")
 		item = create_item("_Test Item 1")
+		create_customer(customer_name="_Test Customer",company="_Test Company")
 		shipping_rule = create_shipping_rule(
 			shipping_rule_type="Selling", 
 			shipping_rule_name="Shipping Rule - Test Fixed",

